mlflow/cli.py

```diff
@@ -73,7 +73,7 @@ def cli():
 def run(uri, entry_point, version, param_list, experiment_id, mode, cluster_spec, git_username,
```

Comment:
Any reason we didn't want to add control over blocking as an option? It may be useful even for people using the shell. We can add `--sync/async` or `--block/no-block` or something like that.

Reply:
Agreed it would be nice to be able to launch a run with e.g. `--async` and get back e.g. the run ID on stdout. Maybe that's a reason to use `subprocess.Popen` as described [here](https://github.com/databricks/mlflow/pull/82#discussion_r200243643)